mlir/lib/Dialect/Rock/Generator/Conv2dGenerator.cpp

```diff
@@ -337,20 +339,30 @@ int Conv2dGenerator::getBwdDataKernelCount() const {
   return static_cast<int>(gemmIds.size());
 }
 
-Type Conv2dGenerator::getDataType(OpBuilder &builder) const {
+static Type strToType(std::string dataTypeStr, OpBuilder &builder) {
```

Comment:
While we're here, can we make this a `llvm::StringSwitch` and make the argument a `StringRef`?